circuits/integration/zk_circuits/valid_match_mpc.rs

```diff
@@ -204,47 +174,42 @@ fn setup_witness<N: MpcNetwork + Send, S: SharedValueSource<Scalar>>(
         balance2: allocated_balance2,
         price2: allocated_price2,
         match_res,
-    })
+    }
 }
 
 /// Prove and verify a valid match MPC circuit, return `true` if verification succeeds
-fn prove_and_verify_match<N: MpcNetwork + Send, S: SharedValueSource<Scalar>>(
-    witness: AuthenticatedValidMatchMpcWitness<N, S>,
-    fabric: SharedFabric<N, S>,
+fn prove_and_verify_match(
+    witness: AuthenticatedValidMatchMpcWitness,
+    fabric: MpcFabric,
 ) -> Result<bool, String> {
     // Prove
     let (witness_comm, proof) =
-        multiprover_prove::<'_, _, _, ValidMatchMpcCircuit<'_, _, _>>(witness, (), fabric)
+        multiprover_prove::<'_, ValidMatchMpcCircuit<'_>>(witness, (), fabric)
             .map_err(|err| format!("Error proving: {:?}", err))?;
 
     // Open
-    let opened_proof = proof
-        .open()
-        .map_err(|err| format!("Error opening proof: {:?}", err))?;
-    let opened_comm = witness_comm
-        .open_and_authenticate()
+    let opened_proof =
+        await_result(proof.open()).map_err(|err| format!("Error opening proof: {:?}", err))?;
```

Comment:
nit: could just `await_result_with_error` here (and elsewhere were a generic `map_err` is attached to `await_result`)